core/src/main/scala-2/chisel3/internal/SourceInfoFileResolver.scala

```diff
@@ -13,13 +13,18 @@ import scala.reflect.macros.blackbox.Context
 
 /** Scala compile-time function for determine the String used to represent a source file path in SourceInfos */
 private[internal] object SourceInfoFileResolver {
+  // Ensure non-empty paths end in a slash
+  def sanitizePath(path: String): String = {
+    if (path.isEmpty) path
+    else if (path.last != '/') path + '/'
+    else path
+  }
```

Comment:
This feels janky. Is there some formulation that involves a native Java library or, at minimum, `java.io.File.separator` so that our Windows friends aren't left in the cold?

Reply:
Good point, I will use `File.separatorChar`.